src/librbd/io/ObjectRequest.cc

```diff
@@ -246,6 +253,22 @@ void ObjectReadRequest<I>::handle_read_object(int r) {
     return;
   }
 
+  // merge ExtentResults to a single sparse bufferlist
+  int pos = 0;
+  for (auto& [read_data, extent_map] : m_extent_results) {
+    auto read_data_ptr = read_data.get();
+    auto extent_map_ptr = extent_map.get();
+    m_read_data->append(*read_data_ptr);
```

Comment:
Nit: slightly more efficient to `claim_append` -- just need to cache your data length first since you use it below